var/spack/repos/builtin/packages/sundials/package.py

```diff
@@ -27,138 +27,531 @@
 import sys
 
 
-class Sundials(Package):
+class Sundials(CMakePackage):
     """SUNDIALS (SUite of Nonlinear and DIfferential/ALgebraic equation
     Solvers)"""
 
-    homepage = "http://computation.llnl.gov/casc/sundials/"
-    url = "http://computation.llnl.gov/projects/sundials-suite-nonlinear-differential-algebraic-equation-solvers/download/sundials-2.6.2.tar.gz"
+    homepage = "https://computation.llnl.gov/projects/sundials"
+    url = "https://computation.llnl.gov/projects/sundials/download/sundials-2.7.0.tar.gz"
 
-    version('2.7.0', 'c304631b9bc82877d7b0e9f4d4fd94d3')
+    # ==========================================================================
+    # Versions
+    # ==========================================================================
+
+    version('3.0.0-beta-2',
+            git='https://github.com/LLNL/sundials.git', tag='v3.0.0-beta-2')
+    version('2.7.0', 'c304631b9bc82877d7b0e9f4d4fd94d3', preferred=True)
     version('2.6.2', '3deeb0ede9f514184c6bd83ecab77d95')
 
+    # ==========================================================================
+    # Variants
+    # ==========================================================================
+
+    # SUNDIALS solvers
+    variant('cvode',  default=True,
+            description='Enable CVODE')
+    variant('cvodes', default=True,
+            description='Enable CVODES')
+    variant('arkode', default=True,
+            description='Enable ARKode')
+    variant('ida',    default=True,
+            description='Enable IDA')
+    variant('idas',   default=True,
+            description='Enable IDAS')
+    variant('kinsol', default=True,
+            description='Enable KINSOL')
+
+    # Real precision type
+    variant(
+        'precision',
+        default='double',
+        description='''real type precision''',
+        values=('single', 'double', 'extended'),
+        multi=False
+    )
+
+    # Index type
+    variant(
+        'indextype',
+        default='int64_t',
+        description='''index integer type''',
+        values=('int32_t', 'int64_t'),
+        multi=False
+    )
+
+    # Parallelism
     variant('mpi',     default=True,
-            description='Enable MPI parallelism')
-    variant('lapack',  default=True,
-            description='Use external BLAS/LAPACK libraries')
-    variant('klu',     default=False,
+            description='Enable MPI parallel vector')
+    variant('openmp',  default=False,
```

Comment:
are `openmp` and `pthread` mutually exclusive (i.e. only one can be `True` at the same time)? If so, could you please rework it into a single variant:
```
variant(
        'threads', default='none',
        description='Multithreading support',
        values=('pthreads', 'openmp', 'none'),
        multi=False
    )
```
that's what we do with all `blas/lapac` (i.e. see `openblas`)

Reply:
In our case `openmp` and `pthread` are not mutually exclusive options. Turning on these variants builds separate `openmp` and `pthread` vector modules that can be used together if a user wanted.

Reply:
i see, thanks for explanation